std/range/package.d

```diff
@@ -9594,6 +9594,324 @@ auto refRange(R)(R* range)
     foo(r);
 }
 
+/**
+Bitwise adapter over Integral type Ranges. Consumes the range elements bit by bit.
+*/
+struct Bitwise(R)
+    if (isInputRange!R && isIntegral!(ElementType!R))
+{
+    alias ER = ElementType!R;
+    alias UER = Unsigned!ER;
+
+    private R parent;
+    private UER mask;
+    private int maskPos;
+    private ER accumulator;
+
+    private enum UER bitsNum = ER.sizeof * 8;
+
+    this()(auto ref R range)
+    {
+        parent = range;
+
+        static if (hasLength!R)
+        {
+            len = parent.length * bitsNum;
+        }
+
+        initPrivates();
+
+        static if (isBidirectionalRange!R)
+        {
+            initBackPrivates();
+        }
+    }
```

Comment:
As discuss, this could cause grief to clients because it forces a call to popFront and a call to popBack from the parent, even if this range is not used at all.